### ISSUE TITLE
False positive on TRY203

### ISSUE THREAD

**Issue by Ahajha** [astral-sh/ruff#27775]
### Summary

Repro:
```python
def fail() -> None:
    raise ValueError()


def a() -> None:
    try:
        fail()
    except Exception:
        raise
    finally:
        print("finally")


def b() -> None:
    fail()
    print("finally")


print("a")
try:
    a()
except:
    pass
print("b")
try:
    b()
except:
    pass
```
https://play.ruff.rs/30554c6e-ac10-4521-9805-b1b518b96bed

TRY203 will suggest to rewrite `a()` as `b()`, but this changes behavior. We should noqa this case, but I believe that _any_ try block with a `finally` should not have TRY203 flag it.

### Version

ruff 0.16.3

**Comment by jonathandung** [comment]
The diagnostic message is perfectly clear:

> Remove exception handler; error is immediately re-raised

The "exception handler" here refers to the `except:` block and not the `finally:` block, which I don't believe is ambiguous. There is no auto-fix for this rule either.

**Comment by jonathandung** [comment]
The docs example also doesn't have a cleanup clause, so there's little room for misinterpretation here.

**Comment by Ahajha** [comment]
Oh... I think I see. So it would just be a try/finally block? Is that allowed? I'll have to try and see.

**Comment by Ahajha** [comment]
Yup, you're right, sorry about the noise. I'm used to C++ where `finally` blocks don't exist 😆
